### PR TITLE
Small group_by improvements

### PR DESCRIPTION
**Pull request by coszio**
- improve termination condition check
- add all_ids to aggregator to avoid flat_map
- use ahash for aggregator hashmaps and hashsets

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by coszio** [comment]
Disclaimer: tbh `ahash` seemed to have a only a little performance improvement over regular hashmaps, but the benchmark used int group ids, whereas ahash should shine with longer strings. Still I am not sure if it is enough evidence to switch from std hashmaps

**Comment by agourlay** [comment]
My gut feeling is that we do not need `ahash`.

AFAIK this feature is mostly IO bound, improving on the hashing scheme of the group keys won't yield a significant gain in real life.

I would recommend test driving your improvements at a system level with BFB to see if they are visible end to end.

What kind of improvements have you seen for `improve termination condition check` and `add all_ids to aggregator to avoid flat_map`?

**Comment by generall** [comment]
> My gut feeling is that we do not need ahash.

Agreed, that looks like a preliminary optimization. 
I don't see any evidence, that `HashMap` in this particular case is any kind of bottleneck

**Comment by agourlay** [comment]
I'll do a proper review once `ahash` is removed, it should make the diff easier to digest.

File: lib/collection/src/grouping/group_by.rs [hunk 1]
```diff
@@ -308,7 +304,7 @@ where
         }
 
         // exclude already aggregated points
-        let ids = aggregator.ids();
+        let ids = aggregator.ids().clone();
```
Comment (agourlay) [inline lib/collection/src/grouping/group_by.rs:307]: Why is the `HashSet` cloned?

**Review by agourlay — APPROVED** [review]

**Review by generall — APPROVED** [review]

**Pull request merged**